src/core/net/tcp6.cpp

```diff
@@ -163,25 +232,48 @@ Error Tcp::Endpoint::Deinitialize(void)
     SuccessOrExit(error = tcp.mEndpoints.Remove(*this));
     SetNext(nullptr);
 
+    SuccessOrExit(error = Abort());
+
 exit:
     return error;
 }
 
 uint8_t Tcp::Endpoint::TimerFlagToIndex(uint8_t aTimerFlag)
 {
-    OT_UNUSED_VARIABLE(aTimerFlag);
-    /*
-     * TODO: Convert from the timer flag provided by TCPlp to the index in
-     * our timers array.
-     */
-    return 0;
+    switch (aTimerFlag)
+    {
+    case TT_DELACK:
+        return kTimerDelack;
+    case TT_REXMT:
+        /* fallthrough intentional */
+    case TT_PERSIST:
+        return kTimerRexmtPersist;
+    case TT_KEEP:
+        return kTimerKeep;
+    case TT_2MSL:
+        return kTimer2Msl;
+    default:
+        OT_ASSERT(false);
+        return 0; /* prevents a compiler warning */
```

Comment:
Avoid using multiple `return` statements?

Per our style guide, we need a single `return` from function/method.

> 
>  - There should be one return statement per free function or method at the end of the free function or method.

Reply:
OK. Fixed in the latest update.